test/Algebra/Graph/Test/Generic.hs

```diff
@@ -522,6 +523,25 @@ testHasEdge (Testsuite prefix (%)) = do
         (u, v) <- elements ((x, y) : edgeList z)
         return $ hasEdge u v z == elem (u, v) (edgeList % z)
 
+testHasLoop :: Testsuite -> IO ()
+testHasLoop (Testsuite prefix (%)) = do
+    putStrLn $ "\n============ " ++ prefix ++ "hasSelfLoop ============"
+    test "hasSelfLoop x empty            == False" $ \x ->
+          hasSelfLoop x % empty          == False
+
+    test "hasSelfLoop x (vertex y)       == False" $ \x y ->
+          hasSelfLoop x % vertex y       == False
+
+    test "hasSelfLoop x (edge x x)       == True" $ \x ->
+          hasSelfLoop x % edge x x       == True
+
+    test "hasSelfLoop x . removeEdge x x == const False" $ \x y ->
+         (hasSelfLoop x . removeEdge x x) y == const False % y
+
+    test "hasSelfLoop x                  == elem (x,x) . edgeList" $ \x y -> do
```

Comment:
I think we should replace this test with `hasSelfLoop x == hasEge x x`, which is in the Haddock.